flagship/dist-deno/src/api/BatchingCachingStrategyAbstract.ts

```diff
@@ -0,0 +1,79 @@
+import { IFlagshipConfig } from '../config/index.ts'
+import { HIT_CACHE_VERSION, HIT_DATA_CACHED, HIT_DATA_FLUSHED, PROCESS_CACHE_HIT, PROCESS_FLUSH_HIT } from '../enum/index.ts'
+import { HitAbstract } from '../hit/index.ts'
+import { HitCacheDTO } from '../types.ts'
+import { IHttpClient } from '../utils/HttpClient.ts'
+import { logDebug, logError, sprintf } from '../utils/utils.ts'
+import { ITrackingManagerCommon } from './TrackingManagerAbstract.ts'
+
+export const LOOKUP_HITS_JSON_OBJECT_ERROR = 'JSON DATA must fit the type HitCacheDTO'
+export const LOOKUP_VISITOR_JSON_OBJECT_ERROR = 'JSON DATA must fit the type VisitorCacheDTO'
+
+export abstract class BatchingCachingStrategyAbstract implements ITrackingManagerCommon {
+    protected _config : IFlagshipConfig;
+    protected _hitsPoolQueue: Map<string, HitAbstract>
+    protected _httpClient: IHttpClient;
+
+    public get config () : IFlagshipConfig {
+      return this._config
+    }
+
+    constructor (config: IFlagshipConfig, httpClient: IHttpClient, hitsPoolQueue: Map<string, HitAbstract>) {
+      this._config = config
+      this._hitsPoolQueue = hitsPoolQueue
+      this._httpClient = httpClient
+    }
+
+    abstract addHit (hit: HitAbstract): Promise<void>
+
+    abstract sendBatch(): Promise<void>
+
+    abstract notConsent(visitorId: string): Promise<void>
+
+    protected async cacheHit (hits:Map<string, HitAbstract>):Promise<void> {
+      try {
+        const hitCacheImplementation = this.config.hitCacheImplementation
+        if (this.config.disableCache || !hitCacheImplementation || typeof hitCacheImplementation.cacheHit !== 'function') {
+          return
+        }
+
+        const data : Record<string, HitCacheDTO> = {}
+
+        hits.forEach((item, key) => {
```

Comment:
I would have preferred a .reduce to construct the object, or a Object.fromEntries, but it's more a matter of taste